### DISCUSSION TITLE
Upsert on unique and nullable column

### DISCUSSION THREAD

**Discussion by huynhhacnguyen** [prisma/prisma#9774] [Q&A]
I'm having a model with a unique & nullable column and I'm struggling to create an upsert query.
```prisma
model App {
  id               Int    @id @default(autoincrement())
  createdAt  DateTime? @unique
}
```
If I use null in the where clause, the following error is thrown:
```ts
    const version = await this.dbService.app.upsert({
      where: {
        createdAt: null
      },
      update: {},
      create: {}
    })

Argument createdAt for where.createdAt must not be null. Please use undefined instead.
```
However, if I use undefined instead, another error pops up:
```
Argument where of type AppWhereUniqueInput needs at least one argument. Available args are listed in green.
```
Is there something I'm missing here or is this a limitation of the current version? Any help will be appreciated.
Prisma version: 2.13.0

**Comment by ryands17** [comment]
@huynhhacnguyen  
What would happen if there are multiple records with `null` in this case? `null` is unique in the sense but comparison is always done with `field is NULL` so I don't think you can perform an upsert like this.

**Reply by huynhhacnguyen** [reply]
My app logic here is ensuring that there is only one record with null and multiple records with non-null unique values, but I get your point, since I'm using postgres, and postgres does allow multiple records with null on a unique column. So you're saying there isn't an official solution for this, right? I guess I can totally restructure my model, instead of keeping that record with null I can set it to a special number (epoch time) and still maintain the original logic, just that it sounded like a workaround and I wanted to check if there's a better way. Thanks for the fast reply anw.

**Reply by ryands17** [reply]
Unfortunately there isn't an official solution. You can use the workaround you've specified at the moment, but a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with the above use case would be great 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
